homeassistant/helpers/template.py

```diff
@@ -676,6 +677,43 @@ def __repr__(self) -> str:
         return "<template AllStates>"
 
 
+class StateTranslated:
+    """Class to represent a translated state in a template."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize all states."""
+        self._hass = hass
+
+    def __call__(self, entity_id, language):
```

Comment:
missing types, and should we validate that the language is preloaded and generate an error if not?

Reply:
Types added.

It's possible to use language that is not preloaded via yaml config as long as it has been loaded by frontend. Current implementation returns original state if it's not possible to translate it, but this approach can be changed.